frontend/src/views/RoleView.vue

```diff
@@ -66,24 +71,24 @@ const hideDialog = () => {
 }
 
 const createKnowledge = async () => {
-  if (newKnowledgeName.value == '') {
+  if (knowledge.value.name == '') {
     ErrorMessage('You need to fill in this field')
     return
   }
   try {
-    /*await knowledgesStore.saveKnowledge(
-      name: newKnowledgeName.value,
-      roleId: '1',
-      levelId: '1',
-      weight: '1'
-    )*/
+    await knowledgesStore.saveKnowledge({
```

Comment:
Why not pass the knowledge here? Like this
```suggestion
    await knowledgesStore.saveKnowledge(knowledge)
```